common/src/chain/tokens/nft.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crypto::key::PublicKey;
+use serialization::{extras::non_empty_vec::DataOrNoVec, Decode, Encode};
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode)]
+pub struct NftIssuanceV1 {
+    pub metadata: Metadata,
+    // TODO: Implement after additional research payout, royalty and refund.
+    //       Payout might be Multisig contract with amount enforcement.
+}
+
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Encode,
+    Decode,
+    serde::Serialize,
+    serde::Deserialize,
+)]
+pub struct TokenCreator {
+    pub public_key: PublicKey,
+}
+
+impl From<PublicKey> for TokenCreator {
+    fn from(public_key: PublicKey) -> Self {
+        Self { public_key }
+    }
+}
+
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Encode,
+    Decode,
+    serde::Serialize,
+    serde::Deserialize,
+)]
+pub struct Metadata {
```

Comment:
You should separate this from RPC by having a special type just for RPC to avoid having this in RPC. Please create a type that's more friendly with RPC instead of implementing serde for everything.

Reply:
Ok, I will do that

Reply:
I added separated types for RPC. But I have token creator field in `RPCNonFungibleTokenMetadata` that contain a `PublicKey`. I think that encoding it to `Vec<u8>` is better than adding `serde` requirements to many types. Anyway, if you have any suggestions how to do it better, let me know

Reply:
Vec for RPC is fine.